packages/wonder-blocks-core/util/add-style.js

```diff
@@ -7,11 +7,12 @@ import {processStyleList} from "./util.js";
 
 import type {StyleType} from "./types.js";
 
-export default function addStyle<T: Object>(
-    Component: React.ComponentType<T> | string,
+// TODO(kevinb): have an a version which uses exact object types
+export default function addStyle<T: React.ComponentType<*> | string>(
+    Component: T,
     defaultStyle?: StyleType,
-): React.ComponentType<T & {style: StyleType}> {
-    function StyleComponent(props: T & {style: StyleType}) {
+): React.ComponentType<React.ElementProps<T> & {style: StyleType}> {
+    function StyleComponent(props: React.ElementProps<T> & {style: StyleType}) {
```

Comment:
Should this be `React.ElementConfig` so that the defaultProps are taken into account?

Reply:
I didn't even realize such a thing existed.  I'll give it a try.